app/models/form.rb

```diff
@@ -47,6 +47,29 @@ def next_page_redirect_path(page, case_log)
     end
   end
 
+  def next_incomplete_section_redirect_path(subsection, case_log)
+    subsection_ids = subsections.map(&:id)
+
+    if case_log.status == "completed"
+      next_subsection = get_subsection(subsection_ids[subsection_ids.length - 1])
+      first_question_in_subsection = next_subsection.pages.first.id
+      return first_question_in_subsection.to_s.dasherize
+    end
+
+    next_subsection_id_index = subsection_ids.index(subsection.id) + 1
+    next_subsection = get_subsection(subsection_ids[next_subsection_id_index])
+
+    if next_subsection.id == "declaration" && case_log.status != "completed"
+      next_subsection = get_subsection(subsection_ids[0])
+    end
+    if next_subsection.status(case_log) == :completed
+      next_incomplete_section_redirect_path(next_subsection, case_log)
+    else
+      first_question_in_subsection = next_subsection.pages.first.id
+      first_question_in_subsection.to_s.dasherize
+    end
+  end
```

Comment:
This is really ugly but I don't have any ideas about how to make it nicer. Open to suggestions or a brief pairing session to tidy this up

Reply:
Summarising previous discussion slightly:

I think part of the reason this is odd is that `sections` don't really have paths, pages do. If a section is incomplete (in progress) but the first question **has** been answered - should we really direct them to that first page anyway or rather to the first unanswered question's page?

I think we want to be finding the next page to direct them to (and there is a method for that, it just doesn't work for when the next page is in a different section). Now that you've got tests for your use case though I think it would make sense to refactor it to do that?